pkg/kubectl/cmd/cmd.go

```diff
@@ -393,13 +393,17 @@ func (h *DefaultPluginHandler) Execute(executablePath string, cmdArgs, environme
 // HandlePluginCommand receives a pluginHandler and command-line arguments and attempts to find
 // a plugin executable on the PATH that satisfies the given arguments.
 func HandlePluginCommand(pluginHandler PluginHandler, cmdArgs []string) error {
-	remainingArgs := []string{} // all "non-flag" arguments
-
-	for idx := range cmdArgs {
-		if strings.HasPrefix(cmdArgs[idx], "-") {
+	var remainingArgs []string // all "non-flag" arguments
+	for _, arg := range cmdArgs {
+		if strings.HasPrefix(arg, "-") {
 			break
 		}
-		remainingArgs = append(remainingArgs, strings.Replace(cmdArgs[idx], "-", "_", -1))
+		remainingArgs = append(remainingArgs, strings.Replace(arg, "-", "_", -1))
+	}
+
+	if len(remainingArgs) == 0 {
+		// the length of cmdArgs is at least 1
+		return fmt.Errorf("flags cannot be placed before plugin name: %s", cmdArgs[0])
```

Comment:
If I invoke `kubectl foo --flag bar` this error will complain about `foo`, but should about `--flag`. I'd suggest to just print the error w/o anything it'll usually look like this:
```
kubectl foo --flag bar
Error: flags cannot be placed before plugin name
```
Alternatively you'd need to look for the first flag, but that's not necessary, imo.

Reply:
If the binary `kubectl-foo` does not exist, I think the following output is expected
```
Error: unknown command "foo" for "kubectl"

Did you mean this?
        top

Run 'kubectl --help' for usage.
```

If `kubectl-foo` does exist, kubectl with this patch should work without complaining.